octopipes/benchmark.py

```diff
@@ -8,14 +8,29 @@
 
 
 class Benchmark:
-    def __init__(self, dataloader: Dataloader, workflows: list[Workflow],
-                 flows_factory: AggregateFlowsFactory | None = None) -> None:
+    def __init__(
+        self,
+        dataloader: Dataloader,
+        workflows: list[Workflow],
+        flows_factory: AggregateFlowsFactory | None = None,
+        mode: str | None = None,
+    ) -> None:
+
+        """
+        :param str mode: "single" for sequential execution, "threaded" for multithreading.
+        """
+
         self.dataloader = dataloader
         self.workflows = workflows
+        self.mode=mode
```

Comment:
Since the mode is chosen based on the dataloader, it's unnecessary to add it to the constructor.